src/glmfit.jl

```diff
@@ -521,6 +528,10 @@ const FIT_GLM_DOC = """
     for a list of built-in links).
 
     # Keyword Arguments
+    - `dropcollinear::Bool=true`: Controls whether or not `lm` accepts a model matrix which
+      is less-than-full rank. If `true` (the default), only the first of each set of
+      linearly-dependent columns is used. The coefficient for redundant linearly dependent columns is
+      `0.0` and all associated statistics are set to `NaN`.
```

Comment:
this statement does not seem to be precise. Consider e.g. the following. `x1, x2, x3, x4, x5` columns (in this order) where `x1`, `x2`, and `x4` are generated randomly and independent, and then `x3=x1+x2` and `x5=x3+x4`.

The implementation drops columns `x3` and `x4`. Reading the description I would expect `x3` and `x5` to be dropped.

@nalimilan - this seems to be unrelated to this PR but a general issue (and probably is a detail that can be resolved later)

Reply:
Good catch. Do you think it's clearly defined what columns will be dropped by the pivoted cholesky decomposition? In your example, why is `x4` dropped and not `x5`?

Reply:
I think it would be also useful to test the case when collinearity is introduced by `@formula` term, e.g. `@formula(y ~ identity(x1+x2)+x1+x2)` (I think nothing bad should happen but I would test it)